force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,9 +200,13 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer generateRandomNumber() {
         // Use Math class to generate number between 50 and 100
-
+        Integer randomNumberGenerated = 0;
+        while(randomNumberGenerated != null && randomNumberGenerated <= 50) {
+            Double randomNum = Math.random();
+            randomNumberGenerated = (Integer)(randomNum * 100);
+        }
```

Comment:
Again I would never have thought of iterating over Math.random. Clever but harder  than you needed to work. 

 return Math.round(Math.random() * 50) + 50;